process/track/blockNotarizer_test.go

```diff
@@ -62,9 +86,69 @@ func TestAddNotarizedHeader_ShouldWork(t *testing.T) {
 	assert.Equal(t, &hdr2, bn.GetNotarizedHeaderWithIndex(0, 1))
 }
 
+func TestCleanupWhenMaxCapacityIsReached_ShouldNotCleanNotarizedHeadersWhenMaxCapacityIsNotReached(t *testing.T) {
+	t.Parallel()
+	bn, _ := track.NewBlockNotarizer(&mock.HasherMock{}, &mock.MarshalizerMock{}, mock.NewMultipleShardsCoordinatorMock())
+
+	maxNumHeadersToKeepPerShard := 1200
+
+	for i := uint64(0); i < uint64(maxNumHeadersToKeepPerShard); i++ {
+		bn.AppendNotarizedHeader(&block.Header{Nonce: i})
+	}
+
+	bn.CleanupWhenMaxCapacityIsReached(0)
+	require.Equal(t, maxNumHeadersToKeepPerShard, len(bn.GetNotarizedHeaders()[0]))
+}
+
+func TestCleanupWhenMaxCapacityIsReached_ShouldCleanOldestNotarizedHeaders(t *testing.T) {
+	t.Parallel()
+	bn, _ := track.NewBlockNotarizer(&mock.HasherMock{}, &mock.MarshalizerMock{}, mock.NewMultipleShardsCoordinatorMock())
+
+	maxNumHeadersToKeepPerShard := 1200
+	numHeadersToRemovePerShard := 200
+	notarizedHeadersCount := maxNumHeadersToKeepPerShard + 1
+	shardID := uint32(0)
+
+	for i := uint64(0); i < uint64(notarizedHeadersCount); i++ {
+		bn.AppendNotarizedHeader(&block.Header{Nonce: i, ShardId: shardID})
+	}
+
+	bn.CleanupWhenMaxCapacityIsReached(shardID)
+	require.Equal(t, maxNumHeadersToKeepPerShard-numHeadersToRemovePerShard, len(bn.GetNotarizedHeaders()[shardID]))
+
+	firstNonce := notarizedHeadersCount - (maxNumHeadersToKeepPerShard - numHeadersToRemovePerShard)
+	require.Equal(t, uint64(firstNonce), bn.GetNotarizedHeaders()[shardID][0].Header.GetNonce())
+
+	lastNonce := notarizedHeadersCount - 1
+	require.Equal(t, uint64(lastNonce), bn.GetNotarizedHeaders()[shardID][len(bn.GetNotarizedHeaders()[shardID])-1].Header.GetNonce())
+}
+
+func TestCleanupWhenMaxCapacityIsReached_ShouldCleanNewestNotarizedHeaders(t *testing.T) {
+	t.Parallel()
+	bn, _ := track.NewBlockNotarizer(&mock.HasherMock{}, &mock.MarshalizerMock{}, mock.NewMultipleShardsCoordinatorMock())
+
+	maxNumHeadersToKeepPerShard := 1200
```

Comment:
also use here the functions you will extract in export_test.

Reply:
Done